sys/kern/src/arch/arm_m.rs

```diff
@@ -1474,16 +1473,15 @@ unsafe extern "C" fn handle_fault(task: *mut task::Task) {
     // switch to a task to run.
     with_task_table(|tasks| {
         let next = match task::force_fault(tasks, idx, fault) {
-            task::NextTask::Specific(i) => i,
+            task::NextTask::Specific(i) => &tasks[i],
             task::NextTask::Other => task::select(idx, tasks),
-            task::NextTask::Same => idx,
+            task::NextTask::Same => &tasks[idx],
         };
 
-        if next == idx {
+        if next as *const _ == &tasks[idx] as *const _ {
```

Comment:
isn't it better to do this comparison by index rather than by address, because writing `&tasks[idx]` will do a bounds check when accessing the task table, which could be avoided if we just compared the integers? OTTOH then we are using an additional register or stack slot to store both the index and the address... :woman_shrugging:

Reply:
As of this commit, we no longer _have_ the index, and `slice::elem_offset` isn't stable. Part of the performance improvement from this change is returning `&mut Task` from `select` instead of an index.

However, I note that we actually derive `idx` from the `task: *mut Task` value passed in from the ISR wrapper. So I will change the code to compare with that.

This code does not appear to generate a novel bounds check in practice, because `&tasks[idx]` winds up being done a lot of other places in this function implicitly.

I'll do a before and after comparison here in a bit, once I'm done addressing the rest of this feedback.

Reply:
FWIW, I have made this change and will fold it into the PR shortly, but it has no impact on the generated code on v7M or v6M.

Reply:
Update: so I actually measured that incorrectly! As shown by the resulting branch not actually building when I uploaded it. Whoops!

This is a net win, it knocks 16 bytes off the kernel by eliminating one redundant check.

Reply:
If you have the time to answer or perhaps have a ready link available: Are there any flags I can turn on to eg. get debug info included in Hubris builds? Or other ways to make viewing the assembly output easier? Running gdb raw on the output of `cargo xtask dist app.toml` proved to be somewhat convoluted a way to do so, especially with it being a 50/50 chance of `info address func` and `disass func` working or not.